docs/api/field_type_form_and_template.md

```diff
@@ -0,0 +1,194 @@
+
+### Template variables
+
+The block can receive the following variables:
+
+| Name | Type | Description |
+|------|------|-------------|
+| `field` | `eZ\Publish\API\Repository\Values\Content\Field` | The field to display |
+| `contentInfo` | `eZ\Publish\API\Repository\Values\Content\ContentInfo` | The ContentInfo of the Content item the Field belongs to |
+| `versionInfo` | `eZ\Publish\API\Repository\Values\Content\VersionInfo` | The VersionInfo of the Content item the Field belongs to |
+| `fieldSettings` | mixed | Settings of the Field (depends on the Field Type) |
+| `parameters` | hash | Options passed to `ez_render_field()` under the parameters key |
+| `attr` | hash | The attributes to add the generate the HTML markup. Contains at least a class entry, containing <fieldtypeidentifier>-field |
+
+### Reusing blocks
+
+For easier Field Type template development you can take advantage of all defined blocks by using the [`block()` function](http://twig.sensiolabs.org/doc/functions/block.html).
+
+You can for example use `simple_block_field`, `simple_inline_field` or `field_attributes` blocks provided in [`content_fields.html.twig`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.0/eZ/Bundle/EzPublishCoreBundle/Resources/views/content_fields.html.twig#L496).
+
+!!! caution
+
+    To be able to reuse built-in blocks, your template must inherit from `EzPublishCoreBundle::content_fields.html.twig`.
+
+### Registering your template
+
+To make your template available, you must register it in the system.
+
+``` yaml
+ezpublish:
+    system:
+        <siteaccess>:
+            field_templates:
+                -
+                    template: 'AcmeExampleBundle:fields:custom_field_template.html.twig'
+                    # Priority is optional (default is 0). The higher it is, the higher your template gets in the list.
+                    priority: 10
+```
+
+## Back Office templates
+
+### Back Office view template
+
+You can use a separate template for previewing the Field in the Back Office by specifying:
+
+``` yaml
+ezpublish:
+    systems:
+        admin_group:
+            field_templates:
+                - { template: 'AcmeExampleBundle:adminui/field:custom_field_view.html.twig', priority: 10 }
+```
+
+### Field edit template
+
+To use a template for the Field edit form in the Back Office, you need to specify it in configuration
+under the `twig.form_themes` key:
+
+``` yaml
+twig:
+    form_themes:
+        - 'AcmeExampleBundle:adminui/field:custom_field_template.html.twig'
+```
+
+We encourage using custom form types for encapsulation as this makes templating easier by providing Twig block name.
+All built-in Field Types are implemented with this approach. In that case overriding form theme can be done with:
+
+``` html+twig
+{% block custom_fieldtype_widget %}
+    Hello world!
+    {{ block('form_widget') }}
```

Comment:
This calls for more doc about the available variables, the same like in the `Template variables` section. Someone from the Community asked about this, so it's not that obvious. I need to check this along with the previous comment. Some other devs help would be much appreciated.

Reply:
I'll add it to the "todo in future" list in PR description, so as not to bloat this PR any further